src/devices/usbCamera/README.md

```diff
@@ -0,0 +1,287 @@
+
+## 1.7. V4L parameters that can be used together UltraPython
+
+Currently exposed parameters:
+|Name|Code|Default|Min|Max|Note|Status|
+|-|-|-|-|-|-|-|
+|Gain|0x00980913|1|1|11|mapped to a combination of digital and analog gain of the board|Working|
+|Exposure/shutter|0x0098cb03|20msec|1msec|100000msec|limited between 15 msec and 100msec mapped on **tag_l**|Working|
+|White balance|0x0098c9a3-0x0098c9a4|50|0|100|mapped to to the read and blue gain|Working|
+|Brightness|0x0098c9a1|50|0|100|mapped to to the read and blue gain|Working|
+|Subsampling|0x0098cc01|0|0|1|1==subsampling, via specific API|Working|
+
+Internal parameters setted by default:
+|Name|Code|Default|Min|Max|Note|Status|
+|-|-|-|-|-|-|-|
+|ext_trigger|0x0098cc03|1|0|1|Need to be set to 1|Working|
+|tag_h|0x0098cb02|8msec|1msec|100000msec|Dead time between exposures|Working|
+
+Only manual parameters are available for now no auto settings.  
+_Note_ that can be accepted parameters normalized between 0-1 or absolute value. FrameGrabberGu2 send normalized (0-1) parameters.
+
+## 1.8. FPS (frames per second)
+
+It is possible to specify the desired FPS, however FPS has a relation with the exposure.
+
+`Max_Exposure=(1/FPS-8) msec`
+
+The following table is calculated.
+
+| FPS | period msec | Max Exposition in msec |
+| --- | ----------- | ---------------------- |
+| 5   | 200         | 0.192                  |
+| 10  | 100         | 0.100                  |
+| 15  | 66          | 0.066                  |
+| 20  | 50          | 0.050                  |
+| 25  | 40          | 0.040                  |
+| 30  | 33          | 0.033                  |
+| 35  | 28          | 0.028                  |
+| 40  | 25          | 0.025                  |
+| 45  | 22          | 0.022                  |
+
+- if `honorfps` is true the max exposition in mandatory
+- the yarpdev parameter `period` is used by YARP to sample images from the drive at the given period in msec
+
+# 2. usbcamera driver SW modifications
+
+The software follows the c++17 standard.  
+To minimize modifications in the old code and to keep separate old and new cameras code, we create a new class `PythonCameraHelper`. All the UltraPython camera functionalities are developed inside of it.
+If necessary, the class is instantiated by the driver.  
+New PythonCameraHelper class in UML class diagram:
+
+<img src="img/UML004.png" width="600px">
+
+An **dependency injection technique** is used to keep driver and UltraPython camera code separate, so
+test and use of the class in other environment, _are easier_.  
+<img src="img/UML002.png" width="400px">
+
+## 2.1. PytonCameraHelper Cmake options
+
+The following CMake option should be used to compile UltraPython device:
+
+```
+ENABLE_yarpmod_usbCamera        ON
+ENABLE yarpmod_usbCameraRaw     ON
+```
+
+The only new options for UltraPython are:
+
+```
+YARP_USE_UDev                    ON
```

Comment:
Although this README seems to suggest that UDev is specific to UltraPython, it is actually a hard dependency for the whole usbCamera device (see CMakeLists.txt), regardless of the underlying camera model. Is this intentional?

Reply:
Also, `YARP_USE_UDev` doesn't sound very expressive and self-explaining, if it's meant to be specific to the UltraPython.

Better macros? `YARP_USE_UltraPython` looks better in this sense @triccyx.

Reply:
Well, YARP_USE_UDev is not related directly to UltraPyton but to libudev. This library is a standard Linux library for device management. I would not change the macro name as the library could be used for another purpose. @PeterBowman is it possible, with some effort and some #define, to link the use of this library only to UltraPython.. but the correct way to proceed should be to use the library also for USB device.